### PR TITLE
chore(golangci-lint): add thelper linter

### PR DESCRIPTION
**Pull request by Matheusafonsouza**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Enable thelper linter.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2031

Reference umbrella issue [#2031](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2031)

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by Matheusafonsouza** [comment]
I signed it

**Comment by sonasingh46** [comment]
@Matheusafonsouza -- Can you please fix CLA and rebase to main? This is the link which you can follow up to sign up for CLA. 
This should fix the failing jobs.

File: go.mod [hunk 1]
```diff
@@ -103,6 +103,7 @@ require (
 	github.com/inconshreveable/mousetrap v1.0.0 // indirect
 	github.com/josharian/intern v1.0.0 // indirect
 	github.com/json-iterator/go v1.1.12 // indirect
+	github.com/kulti/thelper v0.5.1 // indirect
```
Comment (sonasingh46) [inline go.mod:106]: I am not sure why `thelper` has come here as an indirect dependency. Try removing that and run `go mod tidy`

**Comment by Matheusafonsouza** [comment]
> Can you please fix CLA and rebase to main? This is the link which you can follow up to sign up for CLA

What is the link? you didn't put it on the message

**Comment by Matheusafonsouza** [comment]
Signed CLA :)

**Comment by sonasingh46** [comment]
Sure. Thank you. I missed to put the link. Sorry about that.

**Comment by Matheusafonsouza** [comment]
Hey @sonasingh46 , i rebase to the main but the tests still fail (and it does not makes sense for me that lint is the reason), could you help me?

**Comment by sonasingh46** [comment]
@Matheusafonsouza -- You have still checked in the go.sum file. 
You can delete the go.sum file and run `go mod tidy`. It should fix the issue. 

Or simply do not check in the go.sum file.

**Comment by Matheusafonsouza** [comment]
@sonasingh46 fixed! thank u so much for the help :))

**Comment by jackfrancis** [comment]
@Matheusafonsouza can you kindly squash commits? Thanks!

**Comment by mboersma** [comment]
@Matheusafonsouza, thanks! Could you update the description for this PR here on GitHub and add this line?

Fixes #2031

**Comment by sonasingh46** [comment]
Looks good to me. Please incorporate the comments suggested by @jackfrancis and @mboersma 
Thanks for the PR.

**Comment by Matheusafonsouza** [comment]
Hey @sonasingh46 , @mboersma and @jackfrancis i`m working on it right now, sorry for the delay
i was without network for the past 3 days

**Comment by Matheusafonsouza** [comment]
@sonasingh46 @mboersma @jackfrancis done!

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Comment by jackfrancis** [comment]
@devigned @shysank feel like approving this one?

**Pull request merged**